src/web/client/test/integration/errorHandler.test.ts

```diff
@@ -0,0 +1,634 @@
+        const entity = "";
+        const entityId = "3";
+        //Action
+        const result = checkMandatoryPathParameters(appName, entity, entityId);
+        //Assert
+        const detailMessage = localize(
+            "microsoft-powerapps-portals.webExtension.init.workspace.error.desc",
+            "Check the URL and verify the parameters are correct"
+        );
+        const options = { detail: detailMessage, modal: true };
+        expect(result).false;
+        assert.calledOnceWithExactly(
+            _mockShowErrorDialog,
+            localize(
+                "microsoft-powerapps-portals.webExtension.init.workspace.error",
+                "There was a problem opening the workspace"
+            ),
+            options
+        );
+        assert.calledOnceWithExactly(
+            _mockSendErrorTelemetry,
+            telemetryEventNames.WEB_EXTENSION_MANDATORY_PATH_PARAMETERS_MISSING
+        );
+    });
+
+    it("checkMandatoryPathParameters_shouldReturnFalseWhenEntityIdIsBlank", () => {
+        //Act
+        const _mockShowErrorDialog = spy(vscode.window, "showErrorMessage");
+        const _mockSendErrorTelemetry = spy(
+            powerPlatformExtensionContext.telemetry,
+            "sendErrorTelemetry"
+        );
+        const appName = "portal";
+        const entity = "entity";
+        const entityId = "";
+        //Action
+        const result = checkMandatoryPathParameters(appName, entity, entityId);
+        //Assert
+        const detailMessage = localize(
+            "microsoft-powerapps-portals.webExtension.init.workspace.error.desc",
+            "Check the URL and verify the parameters are correct"
+        );
+        const options = { detail: detailMessage, modal: true };
+        expect(result).false;
+        assert.calledOnceWithExactly(
+            _mockShowErrorDialog,
+            localize(
+                "microsoft-powerapps-portals.webExtension.init.workspace.error",
+                "There was a problem opening the workspace"
+            ),
+            options
+        );
+        assert.calledOnceWithExactly(
+            _mockSendErrorTelemetry,
+            telemetryEventNames.WEB_EXTENSION_MANDATORY_PATH_PARAMETERS_MISSING
+        );
+    });
+
+    it("checkMandatoryPathParameters_shouldReturnFalseWhenPppNameIsDifferentFromPortal", () => {
+        //Act
+        const _mockShowErrorMessage = stub(vscode.window, "showErrorMessage");
+        const appName = "por";
+        const entity = "entity";
+        const entityId = "";
+        //Action
+        const result = checkMandatoryPathParameters(appName, entity, entityId);
+        //Assert
+        const detailMessage = localize(
+            "microsoft-powerapps-portals.webExtension.init.app-not-found",
+            "Unable to find that app"
+        );
+        expect(result).false;
+        assert.calledWith(_mockShowErrorMessage, detailMessage);
+        assert.calledOnce(_mockShowErrorMessage);
```

Comment:
Why you need to mock error message if appName is not portal. It will only return false in default section

Reply:
yes but we are asserting the the function calling "_mockShowErrorMessage" without mock we cannot asserts